substrate-node/pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -922,7 +926,19 @@ impl<T: Config> Pallet<T> {
     }
 
     fn bill_contract_using_signed_transaction(contract_id: u64) -> Result<(), Error<T>> {
+        let author = <pallet_authorship::Pallet<T>>::author();
         let signer = Signer::<T, <T as pallet::Config>::AuthorityId>::any_account();
+
+        // Only allow the author of the block to trigger the billing
+        let signed_message = signer.sign_message(&[0]);
```

Comment:
How did you check if it works?

Reply:
I ran the integration tests and checked the logs as well as the blocks on polkadotjs